src/ledger/LedgerCloseMetaFrame.cpp

```diff
@@ -145,28 +146,22 @@ LedgerCloseMetaFrame::populateTxSet(TxSetXDRFrame const& txSet)
 
 void
 LedgerCloseMetaFrame::populateEvictedEntries(
-    LedgerEntryChanges const& evictionChanges)
+    std::pair<std::vector<LedgerKey>, std::vector<LedgerEntry>> const&
+        evictedEntries)
 {
     releaseAssert(mVersion == 1);
-    for (auto const& change : evictionChanges)
+    for (auto const& key : evictedEntries.first)
     {
-        switch (change.type())
-        {
-        case LEDGER_ENTRY_CREATED:
-            throw std::runtime_error("unexpected create in eviction meta");
-        case LEDGER_ENTRY_STATE:
-            continue;
-        case LEDGER_ENTRY_UPDATED:
-            // The scan also updates the eviction iterator, but should only
-            // update the eviction iterator
-            releaseAssert(change.updated().data.type() == CONFIG_SETTING);
-            continue;
-        case LEDGER_ENTRY_REMOVED:
-            auto const& key = change.removed();
-            releaseAssert(isTemporaryEntry(key) || key.type() == TTL);
-            mLedgerCloseMeta.v1().evictedTemporaryLedgerKeys.push_back(key);
-            break;
-        }
+        releaseAssertOrThrow(isTemporaryEntry(key) || key.type() == TTL);
+        mLedgerCloseMeta.v1().evictedTemporaryLedgerKeys.emplace_back(key);
+    }
+    for (auto const& entry : evictedEntries.second)
+    {
+        releaseAssertOrThrow(isPersistentEntry(entry.data));
+        // Unfortunately, for legacy purposes, evictedTemporaryLedgerKeys is
+        // misnamed and stores all evicted keys, both temp and persistent.
+        mLedgerCloseMeta.v1().evictedTemporaryLedgerKeys.emplace_back(
```

Comment:
Are we still planning to rename this field?

Reply:
Yes, but I haven't pushed the XDR changes yet.